From 374bc67a48e9be326a1ecfec01954a09c115a251 Mon Sep 17 00:00:00 2001
From: Nick Bofferding <bofferdn@us.ibm.com>
Date: Wed, 19 Oct 2016 17:53:41 -0500
Subject: [PATCH] Remove simulation check around untrusted BAR verification

- Remove simics check around verifying untrusted BARs
- Removed various simics workarounds
- Added simics workaround to pull in the untrusted BAR support

Change-Id: Ib1cfba89371eb7ee2ee7b0d7c7ac89969ccb45f3
CMVC-Prereq: 1008942
---
 src/build/citest/etc/workarounds.presimsetup | 6 ++----
 src/usr/hwas/hostbootIstep.C                 | 9 +--------
 2 files changed, 3 insertions(+), 12 deletions(-)

diff --git a/src/build/citest/etc/workarounds.presimsetup b/src/build/citest/etc/workarounds.presimsetup
index 7352a8f..c9760b8 100755
--- a/src/build/citest/etc/workarounds.presimsetup
+++ b/src/build/citest/etc/workarounds.presimsetup
@@ -49,8 +49,6 @@ grep -v "GFW_P8_HB_UNSECURE_OFFSET" \
     $sb/simu/configs/P8_VENICE.config
 echo "SETENV GFW_P8_HB_UNSECURE_OFFSET 58720256" >> \
     $sb/simu/configs/P8_VENICE.config
-echo "Applying workaround to support setting security bits"
-sbex -t 1000007 -r fips860
-echo "Applying SIMICS workaround to allow header copy in secure mode"
-sbex -t 996855 -r fips860
+echo "Applying SIMICS workaround to support secure BARs"
+sbex -t 1008942 -r fips860
 
diff --git a/src/usr/hwas/hostbootIstep.C b/src/usr/hwas/hostbootIstep.C
index 965a569..af37702 100644
--- a/src/usr/hwas/hostbootIstep.C
+++ b/src/usr/hwas/hostbootIstep.C
@@ -77,7 +77,6 @@
 
 #ifdef CONFIG_SECUREBOOT
 #include <secureboot/service.H>
-#include <util/misc.H>
 #endif
 
 #ifdef CONFIG_ENABLE_CHECKSTOP_ANALYSIS
@@ -471,13 +470,7 @@ void* host_prd_hwreconfig( void *io_pArgs )
     {
         #ifdef CONFIG_SECUREBOOT
         // check the BAR values for all processors
-        // TODO RTC 156484 Remove the simics check below from around the BAR
-        // check as soon as simics has support for the BAR value registers.
-        // Also will need to remove include <util/misc.H> from top of this file.
-        if (!Util::isSimicsRunning())
-        {
-            enforce_bar_values_for_all_proc();
-        }
+        enforce_bar_values_for_all_proc();
         #endif
 
         // Flip the scom path back to FSI in case we enabled IBSCOM previously
-- 
1.8.2.2

